### PR TITLE
ProfileManager: Save changes when they happen

### PR DESCRIPTION
**Pull request by fritsch**
Follow-Up of: #16987

### CODE REVIEW

File: xbmc/profiles/ProfileManager.cpp [hunk 1]
```diff
@@ -129,8 +129,7 @@ void CProfileManager::OnSettingsLoaded()
 
 void CProfileManager::OnSettingsSaved() const
```
Comment (Rechi) [inline xbmc/profiles/ProfileManager.cpp]: There is no need to keep this empty function. It has an empty implementation in `ISettingsHandler`.

File: xbmc/profiles/ProfileManager.h [hunk 2]
```diff
@@ -148,14 +148,22 @@ class CProfileManager : protected ISettingsHandler,
     used profile will be loaded
     \return the id to the autologin profile
     */
-  int GetAutoLoginProfileId() const { return m_autoLoginProfile; }
+  int GetAutoLoginProfileId() const
+  {
+    return m_autoLoginProfile;
+    Save();
```
Comment (Rechi) [inline xbmc/profiles/ProfileManager.h]: This function doesn't change any value. Also `Save()` would never get call, as it is after the return statement.
I think you actually wanted to change the `ToggleLoginScreen()` function

**Review by Rechi — COMMENTED** [review]
Have you checked if and when `CProfileManager::OnSettingsCleared` gets called, as that one calls `CProfileManager::Clear`?

File: xbmc/profiles/ProfileManager.cpp (hunk 1 above)
Comment (fritsch) [reply xbmc/profiles/ProfileManager.cpp]: Removed the override. Thanks.

File: xbmc/profiles/ProfileManager.h (hunk 2 above)
Comment (fritsch) [reply xbmc/profiles/ProfileManager.h]: Yes - I was too fast here. I changed it. Thanks much.

**Comment by fritsch** [comment]
It's done when settings shared ptr from application gets out of scope when I see it right. But I did not understand if it's used for some further "Clear() + Save()" functionality, e.g. Reset()

File: xbmc/profiles/ProfileManager.cpp [hunk 3]
```diff
@@ -127,12 +127,6 @@ void CProfileManager::OnSettingsLoaded()
   CDirectory::Create(URIUtils::AddFileToFolder(strDir,"mixed"));
 }
 
-void CProfileManager::OnSettingsSaved() const
-{
-  // save mastercode
-  Save();
-}
```
Comment (fuzzard) [inline xbmc/profiles/ProfileManager.cpp]: This change (thats in leia branch already) has reportedly broken any profile settings being saved

https://github.com/xbmc/xbmc/issues/17412

follow up PR to revert the leia change by user who submitted the issue

https://github.com/xbmc/xbmc/pull/17413

Any insight into why that may be fritsch? Is this really meant to be removed, or an oversight?

**Comment by phunkyfish** [comment]
@fritsch do you want to progress this, close it or put it on the backburner?

**Comment by phunkyfish** [comment]
Where are we with this one?

**Comment by fritsch** [comment]
Not sure - while it is definitely good -  I did not find the root cause of the issue.

**Comment by howie-f** [comment]
@fritsch as far as i can see this is already merged into Leia (#16988). Bad side-effect was that accidently removing `CProfileManager::OnSettingsSaved()` killed saving profile-preferences. i reverted that in #17413. so why now close the whole pr? i'd say bring back `CProfileManager::OnSettingsSaved()`in and merge...no?

**Pull request closed without merging**